[BUG] `bookInfo.id`에서 NaN 확인 오류

### 증상
```ts
TSError: ⨯ Unable to compile TypeScript:
src/books/books.controller.ts:348:23 - error TS2845: This condition will always return 'false'.

348   if (book.id <= 0 || book.id === NaN || bookInfo.id <= 0 || bookInfo.id === NaN) {
                          ~~~~~~~~~~~~~~~

  src/books/books.controller.ts:348:23
    348   if (book.id <= 0 || book.id === NaN || bookInfo.id <= 0 || bookInfo.id === NaN) {
                              ~~~~~~~
    Did you mean 'Number.isNaN(book.id)'?

src/books/books.controller.ts:348:62 - error TS2845: This condition will always return 'false'.

348   if (book.id <= 0 || book.id === NaN || bookInfo.id <= 0 || bookInfo.id === NaN) {
                                                                 ~~~~~~~~~~~~~~~~~~~

  src/books/books.controller.ts:348:62
    348   if (book.id <= 0 || book.id === NaN || bookInfo.id <= 0 || bookInfo.id === NaN) {
                                                                     ~~~~~~~~~~~
    Did you mean 'Number.isNaN(bookInfo.id)'?

```
타입스크립트 오류로 인해 실행할 수 없습니다.

### 스크린샷
![image](https://user-images.githubusercontent.com/54838975/223291050-b4e443aa-b4af-4ed8-b0ee-2b2198aae1ca.png)


### 이유
![image](https://user-images.githubusercontent.com/54838975/223290622-36e157c1-7ced-4663-9abf-3b49e62a56ad.png)

`NaN` !== `NaN` 이기 때문에, 어떤 값이 NaN인지 검사하려면 `Number.isNaN()`을 사용해야 합니다

### 참고

- https://stackoverflow.com/questions/10034149/why-is-nan-not-equal-to-nan
- https://en.wikipedia.org/wiki/Double-precision_floating-point_format
- https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/NaN#설명

### 그밖에
핫픽스 PR을 준비중입니다

--- a/backend/src/books/books.controller.ts
+++ b/backend/src/books/books.controller.ts
@@ -345,7 +345,9 @@ export const updateBookInfo = async (
     status: req.body.status,
   };
 
-  if (book.id <= 0 || book.id === NaN || bookInfo.id <= 0 || bookInfo.id === NaN) { return next(new ErrorResponse(errorCode.INVALID_INPUT, status.BAD_REQUEST)); }
+  if (book.id <= 0 || Number.isNaN(book.id) || bookInfo.id <= 0 || Number.isNaN(bookInfo.id)) {
+    return next(new ErrorResponse(errorCode.INVALID_INPUT, status.BAD_REQUEST));
+  }
   if (!(bookInfo.title || bookInfo.author || bookInfo.publisher || bookInfo.image
           || bookInfo.categoryId || bookInfo.publishedAt || book.callSign || book.status !== undefined)) { return next(new ErrorResponse(errorCode.NO_BOOK_INFO_DATA, status.BAD_REQUEST)); }
 

--- a/backend/src/books/books.controller.ts
+++ b/backend/src/books/books.controller.ts
@@ -4,7 +4,7 @@ import {
 import * as status from 'http-status';
 import * as errorCode from '../utils/error/errorCode';
 import ErrorResponse from '../utils/error/errorResponse';
-import { isNullish } from '../utils/isNullish';
+import isNullish from '../utils/isNullish';
 import { logger } from '../utils/logger';
 import * as BooksService from './books.service';
 import * as types from './books.type';

--- a/backend/src/utils/isNullish.ts
+++ b/backend/src/utils/isNullish.ts
@@ -1,5 +1,3 @@
-export const isNullish = (value: any) => {
-    if (value === null || value === undefined)
-        return (true);
-    return (false);
-  }
+export default function isNullish(value: unknown) {
+  return (value === null || value === undefined);
+}

